service/controller/app/v1/resource/chart/resource.go

```diff
@@ -75,16 +75,27 @@ func (r *Resource) Name() string {
 }
 
 // equals asseses the equality of ReleaseStates with regards to distinguishing fields.
-func equals(a, b *v1alpha1.Chart) bool {
-	if a.Name != b.Name {
+func equals(current, desired *v1alpha1.Chart) bool {
+	if current.Name != desired.Name {
 		return false
 	}
-	if !reflect.DeepEqual(a.Spec, b.Spec) {
+	if !reflect.DeepEqual(current.Spec, desired.Spec) {
 		return false
 	}
-	if !reflect.DeepEqual(a.Labels, b.Labels) {
+	if !reflect.DeepEqual(current.Labels, desired.Labels) {
 		return false
 	}
+
+	desiredAnnotation := desired.GetAnnotations()
+	for currentAnnotationKey, currentValue := range current.GetAnnotations() {
+		if desiredValue, ok := desiredAnnotation[currentAnnotationKey]; ok {
```

Comment:
```suggestion
		if desiredValue, ok := desiredAnnotations[currentAnnotationKey]; ok {
```

As above. Plural makes the code more readable for me.